_episodes/09-cmorization.md

```diff
@@ -0,0 +1,1053 @@
+probably fail, but that's okay. As you will see, the error messages will come in
+handy. Moreover, the recipe will serve as a test case. Once we get it to run, we
+know we have completed our task.
+
+> ## Create a test recipe
+>
+> Create a simple recipe that loads the "FLUXCOM" data. It should include a
+> datasets section with a single entry for the "FLUXCOM" dataset with the correct
+> dataset keys, and a diagnostics section with two variables: gpp and gppStderr.
+> We won't need any preprocessors or scripts (set `scripts: null`), but you will
+> have to add a documentation section with a description, authors and
+> maintainer, otherwise the recipe will fail.
+>
+> Use the following dataset keys:
+>
+> - project: OBS6
+> - dataset: FLUXCOM
+> - type: reanaly
+> - version: ANN-v1
+> - mip: Lmon
+> - start_year: 2000
+> - end_year: 2000
+> - tier: 3
+>
+> Some of these dataset keys are further explained in the callout boxes in this
+> episode.
+>
+> > ## Answer
+> >
+> > Here's an example recipe
+> >
+> > ```yaml
+> > documentation:
+> >
+> >   description: Test recipe for FLUXCOM data
+> >
+> >   authors:
+> >     - kalverla_peter
+> >
+> >   maintainer:
+> >     - kalverla_peter
+> >
+> > datasets:
+> >   - {project: OBS6, dataset: FLUXCOM, mip: Lmon, tier: 3, start_year: 2000, end_year: 2000, type: reanaly, version: ANN-v1}
+> >
+> > diagnostics:
+> >   FLUXCOM:
+> >     description: Check that ESMValTool can load the cmorized FLUXCOM data without errors.
+> >     variables:
+> >       gpp:
+> >       gppStderr:
+> >     scripts: null
+> >
+> > ```
+> >
+> > Note: a recipe similar to this one is available under
+> > `~/path/to/ESMValTool/esmvaltool/recipes/examples/recipe_check_obs.yml`.
+> > That recipe includes checks all datasets for which CMORizers are available.
+> >
+> {: .solution}
+{: .challenge}
+
+Try to run the example recipe with
+
+```bash
+esmvaltool run recipe_check_fluxcom.yml --log_level debug
+```
+
+The `log_level` flag ensures that all relevant information is included in the
+output. We will need this information to find out what needs to be done. Look
+carefully through the log messages. You'll probably find something like
+
+```
+DEBUG   Retrieving OBS6 configuration
+DEBUG   Skipping non-existent /home/peter/default_inputpath/Tier3/FLUXCOM
+DEBUG   Looking for files matching ['OBS6_FLUXCOM_reanaly_ANN-v1_Lmon_gppStderr[_.]*nc'] in []
+ERROR   No input files found for variable {'variable_group': 'gppStderr' ...
+...
+ERROR   Missing data for preprocessor FLUXCOM/gppStderr
+...
+ERROR   Program terminated abnormally, see stack trace below for more information:
+...
+```
+{: .error}
+
+So the output tells us that it cannot find the data, and also where it has been
+looking. Let's try to understand what's going on.
+
+## Location, location, location
+
+The error messages above tell you that ESMValTool cannot find the data. That
+makes total sense, because we have not yet created our CMORized copy of the
```

Comment:
Sorry, I am lost here. I expect this episode shows me how to create a cmorizer script. There are already three well-written episodes on Configuration (episode 3), writing your own recipe (episode 5) and debugging (episode 6). Please avoided duplications and just point the users to those lessons. I see that the debugging exercise here is a nice one, please add it to episode 6.

Reply:
The idea was that the debug messages tell us what we need to fix for the cmorizer. How else would you know what kind of fixes the cmorizer needs to implement? But I can see why you're confused, and I think I know a way to avoid this here.

Reply:
I removed this, but the episode still uses the error messages in a few places. Is it okay like this, or do you still think it's confusing?

Reply:
It is much clear now. Thank you for addressing my comment.